src/test/ui/impl-trait/impl-generic-mismatch-ab.stderr

```diff
@@ -5,7 +5,9 @@ LL |     fn foo<A: Debug>(&self, a: &A, b: &impl Debug);
    |                                -- type in trait
 ...
 LL |     fn foo<B: Debug>(&self, a: &impl Debug, b: &B) { }
-   |                                ^^^^^^^^^^^ expected type parameter, found a different type parameter
+   |            -                   ^^^^^^^^^^^ expected type parameter `B`, found type parameter `impl Debug`
```

Comment:
Calling this "type parameter `impl Debug`" is a little confusing. What about "opaque type `impl Debug`", or something similar instead?

Reply:
That would be reasonable. This output is just a consequence of opaque types being represented as type parameters and having to proactively account for them. Personally I would love to split them into two to avoid a lot of special casing we end up doing, like the one we would have to do here.